From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 1 Jun 2021 08:32:35 +0200
Subject: Revert "kbuild: generate Module.symvers only when vmlinux exists"

This reverts commit d0736af81151cb6213825034da0e09a7e4ad20ea.

diff --git a/.gitignore b/.gitignore
index f53e8be649a24b29178bd99a80e7b69777f736b5..2d334dca54ed4671676b97a4193bc22bf8c401be 100644
--- a/.gitignore
+++ b/.gitignore
@@ -55,7 +55,6 @@ modules.order
 /tags
 /TAGS
 /linux
-/modules-only.symvers
 /vmlinux
 /vmlinux.32
 /vmlinux.symvers
diff --git a/Documentation/dontdiff b/Documentation/dontdiff
index 82e3eee7363b0301a6aaccea965d077130c09f4d..e361fc95ca293d80434ee0cdfae309046c17d612 100644
--- a/Documentation/dontdiff
+++ b/Documentation/dontdiff
@@ -178,7 +178,6 @@ mktables
 mktree
 mkutf8data
 modpost
-modules-only.symvers
 modules.builtin
 modules.builtin.modinfo
 modules.nsdeps
diff --git a/Makefile b/Makefile
index b0e2f455eb87c2ba72b917fc2a6ebcfa75fbecaf..0ff4fbcd8e48e6272f462b599b4755b932d1b917 100644
--- a/Makefile
+++ b/Makefile
@@ -1494,7 +1494,7 @@ endif # CONFIG_MODULES
 # make distclean Remove editor backup files, patch leftover files and the like
 
 # Directories & files removed with 'make clean'
-CLEAN_FILES += include/ksym vmlinux.symvers modules-only.symvers \
+CLEAN_FILES += include/ksym vmlinux.symvers \
 	       modules.builtin modules.builtin.modinfo modules.nsdeps \
 	       compile_commands.json
 
diff --git a/scripts/Makefile.modpost b/scripts/Makefile.modpost
index 42154b6df652918113f499ead1ca16b8c350c183..3c3ccd85213504026b2746eb33af888bc6088c00 100644
--- a/scripts/Makefile.modpost
+++ b/scripts/Makefile.modpost
@@ -65,20 +65,7 @@ else
 ifeq ($(KBUILD_EXTMOD),)
 
 input-symdump := vmlinux.symvers
-output-symdump := modules-only.symvers
-
-quiet_cmd_cat = GEN     $@
-      cmd_cat = cat $(real-prereqs) > $@
-
-ifneq ($(wildcard vmlinux.symvers),)
-
-__modpost: Module.symvers
-Module.symvers: vmlinux.symvers modules-only.symvers FORCE
-	$(call if_changed,cat)
-
-targets += Module.symvers
-
-endif
+output-symdump := Module.symvers
 
 else
 
diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index d191e194167ba73714c8903cbac2631b64182359..1db2bb4180b67de0c1902ef54c01594b2057a518 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -2500,6 +2500,19 @@ static void read_dump(const char *fname)
 	fatal("parse error in symbol dump file\n");
 }
 
+/* For normal builds always dump all symbols.
+ * For external modules only dump symbols
+ * that are not read from kernel Module.symvers.
+ **/
+static int dump_sym(struct symbol *sym)
+{
+	if (!external_module)
+		return 1;
+	if (sym->module->from_dump)
+		return 0;
+	return 1;
+}
+
 static void write_dump(const char *fname)
 {
 	struct buffer buf = { };
@@ -2510,7 +2523,7 @@ static void write_dump(const char *fname)
 	for (n = 0; n < SYMBOL_HASH_SIZE ; n++) {
 		symbol = symbolhash[n];
 		while (symbol) {
-			if (!symbol->module->from_dump) {
+			if (dump_sym(symbol)) {
 				namespace = symbol->namespace;
 				buf_printf(&buf, "0x%08x\t%s\t%s\t%s\t%s\n",
 					   symbol->crc, symbol->name,
