From: Sven Eckelmann <sven@narfation.org>
Date: Fri, 23 Jul 2021 17:22:41 +0200
Subject: ignore missing symbols caused by not compiling whole kernel for headers

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index e1868022ed5067e236bd9c9b7310510328c7c050..e6f07bf8923ae73b33759bea870e94af277ce8c7 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -1746,6 +1746,11 @@ static void check_exports(struct module *mod)
 		const char *basename;
 		exp = find_symbol(s->name);
 		if (!exp) {
+			/* ignore the unresolved symbols caused by only using
+			 * make modules_prepare
+			 */
+			continue;
+
 			if (!s->weak && nr_unresolved++ < MAX_UNRESOLVED_REPORTS)
 				modpost_log(!warn_unresolved,
 					    "\"%s\" [%s.ko] undefined!\n",
