From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 11 Aug 2020 08:08:06 +0200
Subject: Revert "modpost: show warning if vmlinux is not found when processing modules"

This reverts commit 7e8a3235823bcb779acf92de630edd5ddffaf886.

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index 2c80da0220c326efbe4fb7598054cef04ad3566e..fe71a5707b0b9129069b9be06421b35ad6ffba05 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -2399,9 +2399,5 @@ int main(int argc, char **argv)
 		error("Section mismatches detected.\n"
 		      "Set CONFIG_SECTION_MISMATCH_WARN_ONLY=y to allow them.\n");
 
-	if (nr_unresolved > MAX_UNRESOLVED_REPORTS)
-		warn("suppressed %u unresolved symbol warnings because there were too many)\n",
-		     nr_unresolved - MAX_UNRESOLVED_REPORTS);
-
 	return error_occurred ? 1 : 0;
 }
