From: Sven Eckelmann <sven@narfation.org>
Date: Wed, 23 Mar 2022 20:31:00 +0100
Subject: Fix shadow warning from bfb1a7c91fb7 ("x86/bug: Merge annotate_reachable() into _BUG_FLAGS() asm")

diff --git a/arch/x86/include/asm/bug.h b/arch/x86/include/asm/bug.h
index 4d20a293c6fd420952f7897840b06cafe71a1d0c..e1e7add0ae239bd68b44c47abc08411e4cccafa8 100644
--- a/arch/x86/include/asm/bug.h
+++ b/arch/x86/include/asm/bug.h
@@ -78,9 +78,9 @@ do {								\
  */
 #define __WARN_FLAGS(flags)					\
 do {								\
-	__auto_type f = BUGFLAG_WARNING|(flags);		\
+	__auto_type __f = BUGFLAG_WARNING|(flags);		\
 	instrumentation_begin();				\
-	_BUG_FLAGS(ASM_UD2, f, ASM_REACHABLE);			\
+	_BUG_FLAGS(ASM_UD2, __f, ASM_REACHABLE);			\
 	instrumentation_end();					\
 } while (0)
 
