From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 11 Aug 2020 08:08:06 +0200
Subject: Revert "modpost: show warning if vmlinux is not found when processing modules"

This reverts commit 7e8a3235823bcb779acf92de630edd5ddffaf886.

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index b284ee01fdebb90a95213c55b6e99c107de88b08..1c6d765173bd2e061266c30e60a25a7434e3fb90 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -2605,10 +2605,6 @@ int main(int argc, char **argv)
 		}
 	}
 
-	if (nr_unresolved > MAX_UNRESOLVED_REPORTS)
-		warn("suppressed %u unresolved symbol warnings because there were too many)\n",
-		     nr_unresolved - MAX_UNRESOLVED_REPORTS);
-
 	free(buf.p);
 
 	return error_occurred ? 1 : 0;
