From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 21 Apr 2015 19:31:46 +0200
Subject: [PATCH] smatch: Workaround to allow the check of batadv_iv_ogm_process_per_outif
---
 smatch_implied.c | 4 ++--
 smatch_slist.c   | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/smatch_implied.c b/smatch_implied.c
index d33886b..fda70f8 100644
--- a/smatch_implied.c
+++ b/smatch_implied.c
@@ -287,7 +287,7 @@ static void __separate_pools(struct sm_state *sm, int comparison, struct range_l
 	   so we bail.  Theoretically, bailing out here can cause more false
 	   positives but won't hide actual bugs.
 	*/
-	if (sm->nr_children > 4000) {
+	if (sm->nr_children > 40000) {
 		if (option_debug || option_debug_implied) {
 			static char buf[1028];
 			snprintf(buf, sizeof(buf), "debug: %s: nr_children over 4000 (%d). (%s %s)",
@@ -334,7 +334,7 @@ struct sm_state *filter_pools(struct sm_state *sm,
 	if (!sm)
 		return NULL;
 
-	if (sm->nr_children > 4000) {
+	if (sm->nr_children > 40000) {
 		if (option_debug || option_debug_implied) {
 			static char buf[1028];
 			snprintf(buf, sizeof(buf), "debug: %s: nr_children over 4000 (%d). (%s %s)",
diff --git a/smatch_slist.c b/smatch_slist.c
index 62145d2..c0c1398 100644
--- a/smatch_slist.c
+++ b/smatch_slist.c
@@ -239,14 +239,14 @@ int out_of_memory(void)
 	 * It works out OK for the kernel and so it should work
 	 * for most other projects as well.
 	 */
-	if (sm_state_counter * sizeof(struct sm_state) >= 50000000)
+	if (sm_state_counter * sizeof(struct sm_state) >= 500000000)
 		return 1;
 	return 0;
 }
 
 int low_on_memory(void)
 {
-	if (sm_state_counter * sizeof(struct sm_state) >= 25000000)
+	if (sm_state_counter * sizeof(struct sm_state) >= 250000000)
 		return 1;
 	return 0;
 }
