From: Sven Eckelmann <sven@narfation.org>
Date: Wed, 23 Nov 2022 11:02:27 +0100
Subject: Revert "modpost: show warning if any of symbol dump files is missing"

This partially reverts commit 48a0f72797bdc6b428f951aff265f5aecc2bda49.

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/scripts/Makefile.modpost b/scripts/Makefile.modpost
index 39472e834b634009f551dab7bcb72a8b0b70b37f..0516df9ea48b6d6bea4b82cb53fcbc7f722b2d2e 100644
--- a/scripts/Makefile.modpost
+++ b/scripts/Makefile.modpost
@@ -131,12 +131,6 @@ endif # ($(KBUILD_EXTMOD),)
 
 quiet_cmd_modpost = MODPOST $@
       cmd_modpost = \
-	$(if $(missing-input), \
-		echo >&2 "WARNING: $(missing-input) is missing."; \
-		echo >&2 "         Modules may not have dependencies or modversions."; \
-		echo >&2 "         You may get many unresolved symbol errors."; \
-		echo >&2 "         You can set KBUILD_MODPOST_WARN=1 to turn errors into warning"; \
-		echo >&2 "         if you want to proceed at your own risk.";) \
 	$(MODPOST) $(modpost-args)
 
 targets += $(output-symdump)
