From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 1 Jun 2021 08:34:39 +0200
Subject: Revert "kbuild: generate Module.symvers only when vmlinux exists"

This reverts commit 69bc8d386aebbd91a6bb44b6d33f77c8dfa9ed8c.

diff --git a/.gitignore b/.gitignore
index 7afd412dadd2c18b2d1e0c7813b2979ceedabf30..e3ffd1d28194fa5026877231fac87033030e2a1c 100644
--- a/.gitignore
+++ b/.gitignore
@@ -54,7 +54,6 @@ modules.order
 # Top-level generic files
 #
 /linux
-/modules-only.symvers
 /vmlinux
 /vmlinux.32
 /vmlinux.map
diff --git a/Documentation/dontdiff b/Documentation/dontdiff
index 910b30a2a7d98b89ce3b9bc6afb7a88aa7c8025c..ac42ad8d430d1239f8469254639bcd557555ee9b 100644
--- a/Documentation/dontdiff
+++ b/Documentation/dontdiff
@@ -178,7 +178,6 @@ mktables
 mktree
 mkutf8data
 modpost
-modules-only.symvers
 modules.builtin
 modules.builtin.modinfo
 modules.nsdeps
diff --git a/Makefile b/Makefile
index c16a8a2ffd73ef5e83caa226df654f02b9277805..f9f9694b86df87148fd8f29d1b3f1d345d456438 100644
--- a/Makefile
+++ b/Makefile
@@ -1493,7 +1493,7 @@ endif # CONFIG_MODULES
 # make distclean Remove editor backup files, patch leftover files and the like
 
 # Directories & files removed with 'make clean'
-CLEAN_FILES += include/ksym vmlinux.symvers modules-only.symvers \
+CLEAN_FILES += include/ksym vmlinux.symvers \
 	       modules.builtin modules.builtin.modinfo modules.nsdeps \
 	       compile_commands.json .thinlto-cache
 
diff --git a/scripts/Makefile.modpost b/scripts/Makefile.modpost
index 48585c4d04ade8488c27b6e0501d433219b66141..ff517cb924a85cd860d30041d4991092b0c0466d 100644
--- a/scripts/Makefile.modpost
+++ b/scripts/Makefile.modpost
@@ -65,20 +65,7 @@ else
 ifeq ($(KBUILD_EXTMOD),)
 
 input-symdump := vmlinux.symvers
-output-symdump := modules-only.symvers
-
-quiet_cmd_cat = GEN     $@
-      cmd_cat = cat $(real-prereqs) > $@
-
-ifneq ($(wildcard vmlinux.symvers),)
-
-__modpost: Module.symvers
-Module.symvers: vmlinux.symvers modules-only.symvers FORCE
-	$(call if_changed,cat)
-
-targets += Module.symvers
-
-endif
+output-symdump := Module.symvers
 
 else
 
diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index c825372c4d176f16475e6814d079e0bcc8d7c1ac..ca40307ac858a9d29d6688a644b0437ff29a4575 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -2439,6 +2439,19 @@ static void read_dump(const char *fname)
 	fatal("parse error in symbol dump file\n");
 }
 
+/* For normal builds always dump all symbols.
+ * For external modules only dump symbols
+ * that are not read from kernel Module.symvers.
+ **/
+static int dump_sym(struct symbol *sym)
+{
+	if (!external_module)
+		return 1;
+	if (sym->module->from_dump)
+		return 0;
+	return 1;
+}
+
 static void write_dump(const char *fname)
 {
 	struct buffer buf = { };
@@ -2449,7 +2462,7 @@ static void write_dump(const char *fname)
 	for (n = 0; n < SYMBOL_HASH_SIZE ; n++) {
 		symbol = symbolhash[n];
 		while (symbol) {
-			if (!symbol->module->from_dump) {
+			if (dump_sym(symbol)) {
 				namespace = symbol->namespace;
 				buf_printf(&buf, "0x%08x\t%s\t%s\t%s\t%s\n",
 					   symbol->crc, symbol->name,
