From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 11 Aug 2020 08:08:06 +0200
Subject: Revert "modpost: show warning if vmlinux is not found when processing modules"

This reverts commit 7e8a3235823bcb779acf92de630edd5ddffaf886.

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index 96a6e68730981b2403f52f6518903b40d3349864..b3f33d82025d5743a568922cba69054de891f17d 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -2299,9 +2299,5 @@ int main(int argc, char **argv)
 		error("Section mismatches detected.\n"
 		      "Set CONFIG_SECTION_MISMATCH_WARN_ONLY=y to allow them.\n");
 
-	if (nr_unresolved > MAX_UNRESOLVED_REPORTS)
-		warn("suppressed %u unresolved symbol warnings because there were too many)\n",
-		     nr_unresolved - MAX_UNRESOLVED_REPORTS);
-
 	return error_occurred ? 1 : 0;
 }
