From: Sven Eckelmann <sven@narfation.org>
Date: Wed, 23 Nov 2022 11:02:27 +0100
Subject: Revert "modpost: show warning if any of symbol dump files is missing"

This partially reverts commit 48a0f72797bdc6b428f951aff265f5aecc2bda49.

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/scripts/Makefile.modpost b/scripts/Makefile.modpost
index 5eb5e8280379ad9140f0b57bf4c51cae35367279..4cf0f1ff03cf2ecc92a9283ab8d36093e4c53f02 100644
--- a/scripts/Makefile.modpost
+++ b/scripts/Makefile.modpost
@@ -126,10 +126,6 @@ endif
 # Otherwise, allmodconfig would fail with "Argument list too long".
 quiet_cmd_modpost = MODPOST $@
       cmd_modpost = \
-	$(if $(missing-input), \
-		echo >&2 "WARNING: $(missing-input) is missing."; \
-		echo >&2 "         Modules may not have dependencies or modversions."; \
-		echo >&2 "         You may get many unresolved symbol warnings.";) \
 	$(MODPOST) $(modpost-args)
 
 targets += $(output-symdump)
