From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 11 Aug 2020 08:08:06 +0200
Subject: Revert "modpost: show warning if vmlinux is not found when processing modules"

This reverts commit 7e8a3235823bcb779acf92de630edd5ddffaf886.

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index 795b21154446df9d6cd37920e4b5183e0cbddeef..2dd77d57b6eb534433b4d61014821320e12aba8d 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -2293,9 +2293,5 @@ int main(int argc, char **argv)
 		error("Section mismatches detected.\n"
 		      "Set CONFIG_SECTION_MISMATCH_WARN_ONLY=y to allow them.\n");
 
-	if (nr_unresolved > MAX_UNRESOLVED_REPORTS)
-		warn("suppressed %u unresolved symbol warnings because there were too many)\n",
-		     nr_unresolved - MAX_UNRESOLVED_REPORTS);
-
 	return error_occurred ? 1 : 0;
 }
