From: Sven Eckelmann <sven@narfation.org>
Date: Fri, 23 Jul 2021 17:22:41 +0200
Subject: ignore missing symbols caused by not compiling whole kernel for headers

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index 42601eea71e8ef71560b2db92f29ee33fe5d044a..535fd9961c1c59e40b20ffd2e3b3ef159e83f45e 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -2146,6 +2146,11 @@ static void check_exports(struct module *mod)
 		const char *basename;
 		exp = find_symbol(s->name);
 		if (!exp || exp->module == mod) {
+			/* ignore the unresolved symbols caused by only using
+			 * make modules_prepare
+			 */
+			continue;
+
 			if (!s->weak && nr_unresolved++ < MAX_UNRESOLVED_REPORTS)
 				modpost_log(warn_unresolved ? LOG_WARN : LOG_ERROR,
 					    "\"%s\" [%s.ko] undefined!\n",
