From: Sven Eckelmann <sven@narfation.org>
Date: Fri, 23 Jul 2021 17:22:41 +0200
Subject: ignore missing symbols caused by not compiling whole kernel for headers

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index ce56ef8675618f8afa21537bc36b98bee4f6d5ed..74f926419d3a1c55d38d5905e0cd92d20ac20d1c 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -1692,6 +1692,11 @@ static void check_exports(struct module *mod)
 		const char *basename;
 		exp = find_symbol(s->name);
 		if (!exp) {
+			/* ignore the unresolved symbols caused by only using
+			 * make modules_prepare
+			 */
+			continue;
+
 			if (!s->weak && nr_unresolved++ < MAX_UNRESOLVED_REPORTS)
 				modpost_log(!warn_unresolved,
 					    "\"%s\" [%s.ko] undefined!\n",
