From: Sven Eckelmann <sven@narfation.org>
Date: Wed, 23 Nov 2022 11:02:27 +0100
Subject: Revert "modpost: show warning if any of symbol dump files is missing"

This partially reverts commit 48a0f72797bdc6b428f951aff265f5aecc2bda49.

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/scripts/Makefile.modpost b/scripts/Makefile.modpost
index e41dee64d429c42bb3fe3767a24dcdd4157de878..a8a49514a8133f2f9a7e47cb062d6c38adc8293c 100644
--- a/scripts/Makefile.modpost
+++ b/scripts/Makefile.modpost
@@ -115,10 +115,6 @@ MODPOST = scripts/mod/modpost
 # Otherwise, allmodconfig would fail with "Argument list too long".
 quiet_cmd_modpost = MODPOST $@
       cmd_modpost = \
-	$(if $(missing-input), \
-		echo >&2 "WARNING: $(missing-input) is missing."; \
-		echo >&2 "         Modules may not have dependencies or modversions."; \
-		echo >&2 "         You may get many unresolved symbol warnings.";) \
 	sed 's/ko$$/o/' $(or $(modorder-if-needed), /dev/null) | $(MODPOST) $(modpost-args) -T - $(vmlinux.o-if-present)
 
 targets += $(output-symdump)
