From: Sven Eckelmann <sven@narfation.org>
Date: Fri, 23 Jul 2021 17:22:41 +0200
Subject: ignore missing symbols caused by not compiling whole kernel for headers

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index 4e7e37f36756cc574e7132c31003e547760ead85..f01febeabda0414a8f9e395b597c3be65509e85b 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -2096,6 +2096,11 @@ static void check_exports(struct module *mod)
 		const char *basename;
 		exp = find_symbol(s->name);
 		if (!exp) {
+			/* ignore the unresolved symbols caused by only using
+			 * make modules_prepare
+			 */
+			continue;
+
 			if (!s->weak && nr_unresolved++ < MAX_UNRESOLVED_REPORTS)
 				modpost_log(warn_unresolved ? LOG_WARN : LOG_ERROR,
 					    "\"%s\" [%s.ko] undefined!\n",
