From: Sven Eckelmann <sven@narfation.org>
Date: Fri, 23 Jul 2021 17:22:41 +0200
Subject: ignore missing symbols caused by not compiling whole kernel for headers

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index 3c45852db71197ddf7df691c0a75f4d88ea4f6c5..81e147f2fdc84c59cba83dc269337089d41527a2 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -1732,6 +1732,11 @@ static void check_exports(struct module *mod)
 		const char *basename;
 		exp = find_symbol(s->name);
 		if (!exp) {
+			/* ignore the unresolved symbols caused by only using
+			 * make modules_prepare
+			 */
+			continue;
+
 			if (!s->weak && nr_unresolved++ < MAX_UNRESOLVED_REPORTS)
 				modpost_log(warn_unresolved ? LOG_WARN : LOG_ERROR,
 					    "\"%s\" [%s.ko] undefined!\n",
