From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 1 Jun 2021 08:34:39 +0200
Subject: Revert "kbuild: generate Module.symvers only when vmlinux exists"

This reverts commit 69bc8d386aebbd91a6bb44b6d33f77c8dfa9ed8c.

diff --git a/.gitignore b/.gitignore
index 265959544978a795516c321eec0659725a970a90..6c0be0739c3d9284178184ff08fa0f3f043ced80 100644
--- a/.gitignore
+++ b/.gitignore
@@ -55,7 +55,6 @@ modules.order
 # Top-level generic files
 #
 /linux
-/modules-only.symvers
 /vmlinux
 /vmlinux.32
 /vmlinux.map
diff --git a/Documentation/dontdiff b/Documentation/dontdiff
index 352ff53a2306ad897f5ef3e20ade4ca27f664471..c1045433e1ee6ec2a13e690d3e21ead8c0d2bde7 100644
--- a/Documentation/dontdiff
+++ b/Documentation/dontdiff
@@ -178,7 +178,6 @@ mktables
 mktree
 mkutf8data
 modpost
-modules-only.symvers
 modules.builtin
 modules.builtin.modinfo
 modules.nsdeps
diff --git a/Makefile b/Makefile
index d4297b3d0735acbc26724c748ff7a65b9ecb9beb..ecab2f0019c2a557765d34f8c2c63e3f9b1f737d 100644
--- a/Makefile
+++ b/Makefile
@@ -1497,7 +1497,7 @@ endif # CONFIG_MODULES
 # make distclean Remove editor backup files, patch leftover files and the like
 
 # Directories & files removed with 'make clean'
-CLEAN_FILES += include/ksym vmlinux.symvers modules-only.symvers \
+CLEAN_FILES += include/ksym vmlinux.symvers \
 	       modules.builtin modules.builtin.modinfo modules.nsdeps \
 	       compile_commands.json .thinlto-cache
 
diff --git a/scripts/Makefile.modpost b/scripts/Makefile.modpost
index 911606496341ca66a34365dcd29f5cc0d09ff0dd..fc8c742e8bae784d7813ad20d7f465b5e127b975 100644
--- a/scripts/Makefile.modpost
+++ b/scripts/Makefile.modpost
@@ -62,20 +62,7 @@ else
 ifeq ($(KBUILD_EXTMOD),)
 
 input-symdump := vmlinux.symvers
-output-symdump := modules-only.symvers
-
-quiet_cmd_cat = GEN     $@
-      cmd_cat = cat $(real-prereqs) > $@
-
-ifneq ($(wildcard vmlinux.symvers),)
-
-__modpost: Module.symvers
-Module.symvers: vmlinux.symvers modules-only.symvers FORCE
-	$(call if_changed,cat)
-
-targets += Module.symvers
-
-endif
+output-symdump := Module.symvers
 
 else
 
diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index fe71a5707b0b9129069b9be06421b35ad6ffba05..f3942876c59617ac80275c0e269acb2dd8c0fb6b 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -2269,6 +2269,19 @@ static void read_dump(const char *fname)
 	fatal("parse error in symbol dump file\n");
 }
 
+/* For normal builds always dump all symbols.
+ * For external modules only dump symbols
+ * that are not read from kernel Module.symvers.
+ **/
+static int dump_sym(struct symbol *sym)
+{
+	if (!external_module)
+		return 1;
+	if (sym->module->from_dump)
+		return 0;
+	return 1;
+}
+
 static void write_dump(const char *fname)
 {
 	struct buffer buf = { };
